### PR TITLE
fix(legacy) Handle arrayjoin validation

### PR DESCRIPTION
**Pull request by evanh**
Arrayjoin queries are handled differently in the groupby clause. Handle the
difference in the validation.

### CODE REVIEW

File: snuba_sdk/query_visitors.py [hunk 1]
```diff
@@ -288,8 +288,23 @@ def _combine(self, query: "query.Query", returns: Mapping[str, None]) -> None:
...
-                    raise InvalidQuery(f"{group_exp} missing from the groupby")
+                # Array joins work a little differently. In this case the entire arrayjoin
+                # expression shouldn't be in the groupby, just the column that is being
+                # array joined
+                if (
+                    isinstance(group_exp, Function)
+                    and group_exp.function == "arrayJoin"
+                ):
+                    assert group_exp.parameters and isinstance(
+                        group_exp.parameters[0], Column
```
Comment (fpacifici) [inline snuba_sdk/query_visitors.py:299]: Are you sure we cannot arrayJoin over a complex expression? `arrayJoin(f(x))`? It should still work.

Comment (evanh) [reply snuba_sdk/query_visitors.py:299]: This is old. I'll update.

File: snuba_sdk/legacy.py [hunk 2]
```diff
@@ -15,27 +15,43 @@
...
+            return escaped
 
     return value
 
 
 def parse_exp(value: Any) -> Any:
     if isinstance(value, str):
+        if not value or value.startswith("'"):
+            return value.strip("'")
+
```
Comment (fpacifici) [inline snuba_sdk/legacy.py:54]: I am not sure I get what this impact arrayJoin validations.

Comment (evanh) [reply snuba_sdk/legacy.py:54]: These changes are to properly handle the test case I added, which is from the snuba discover tests.

**Review by fpacifici — APPROVED** [review]

**Pull request merged**